staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -350,8 +360,19 @@ var newETCD3Client = func(c storagebackend.TransportConfig) (*clientv3.Client, e
 		TLS:                  tlsConfig,
 		Logger:               etcd3ClientLogger,
 	}
-
-	return clientv3.New(cfg)
+	etcdClient, err := clientv3.New(cfg)
+	if err != nil {
+		return nil, err
+	}
+	consistentListFromCacheEnabled := utilfeature.DefaultFeatureGate.Enabled(features.ConsistentListFromCache)
+	if consistentListFromCacheEnabled {
+		if err := checkVersion(etcdClient, dialTimeout); err != nil {
+			// disable the feature
+			utilfeature.DefaultMutableFeatureGate.OverrideDefault(features.ConsistentListFromCache, false)
+			klog.V(4).Infof("%v", err)
```

Comment:
```suggestion
			klog.ErrorS(err, "Warning: Using ConsistentListFromCache requires newer etcd version, disabling the ConsistentListFromCache feature flag", err)
```

Reply:
Thanks for pointing this out, now we have two errors that both of them are long maybe in runtime, we have `ErrDeprecatedCluster` too maybe look at that.

Reply:
Ups, second `err` was not needed

Reply:
Second option looks good, but we should make it more clear about whether "between" is inclusive or exclusive. Maybe `v3.5.[0-7]`.